sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/AzureKeyVaultKeyWrapProvider.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.encryption;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.encryption.EncryptionUtils;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicReference;
+
+// TODO: moderakh remove blocking calls
+public class AzureKeyVaultKeyWrapProvider implements EncryptionKeyWrapProvider {
```

Comment:
We should add some description to these class.

Reply:
thanks. addressed.